### ISSUE TITLE
trio.open_file() doc confusion

### ISSUE THREAD

**Issue by belm0** [python-trio/trio#614]
It seems trio.open_file() must be used with `async with await ...` rather than `async with ...`.  (Why is that-- can it be improved?  Yury Selivanov calls it an antipattern at https://github.com/python/asyncio/issues/316#issuecomment-183075661)

There is an example in the docs which omits `await`:

```
* Async file objects can be used as async iterators to iterate over
     the lines of the file::

        async with trio.open_file(...) as f:
            async for line in f:
                print(line)
```

https://github.com/python-trio/trio/blame/94a2f8bd082d99a53626309e221da2b348f8c2d7/docs/source/reference-io.rst#L614-L616

**Comment by njsmith** [comment]
I disagree with Yury on this one. (Not sure if current-Yury agrees with 2016-Yury either, that was quite some time ago...)

Trio follows the rule that all you need to know about `await` is that there are async functions and regular functions, and async functions are called like `await some_func()` and regular functions are called like `some_func()`. In fact, the actual `__await__` protocol allows for all kinds of more complicated things – you can `await` objects, you can call async functions without `await` – but we intentionally don't do any of that in order to keep things simple and understandable.

So for example, `open_nursery` is a sync function that returns an async context manager object, so you can use it like `async with open_nursery()`. `open_file` and `open_tcp_stream` are async functions that return trio file and trio stream objects, respectively, so you can use them like `await open_file(...)`, `await open_tcp_stream(...)`. It happens that trio file and trio stream objects both implement the async context manager protocol. So, it follows that you can write `async with await open_file(...)` or `async with await open_tcp_stream(...)`.

But to make both `f = await open_file(...)` work *and* have `async with open_file(...)` work, we'd have to make `open_file` some sort of hybrid function that's sorta-async and sorta-sync and shouldn't even be possible according to the simplified model we teach people in the tutorial. So... we don't do that.

We should fix that example though.
